client/src/modules/tags/modal/createUpdate.js

```diff
@@ -28,6 +28,14 @@ function TagsModalController(
       return false;
     }
 
+    // Do not allow semicolons or vertical bars in tag names since they will be
+    // used to separate the parts of the tags when retrieving them from the back end.
+    if (vm.tags.name.indexOf(';') > -1 || vm.tags.name.indexOf('|') > -1) {
```

Comment:
Clever!

Just as an FYI you can also use [`includes()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes) to detect a character in a string.  It makes it less C-like and more JS like.

Reply:
Done